gcloud-java-logging/src/main/java/com/google/gcloud/logging/spi/v2/ConfigServiceV2Api.java

```diff
@@ -0,0 +1,533 @@
+import com.google.logging.v2.UpdateSinkRequest;
+import com.google.protobuf.Empty;
+import io.gapi.gax.grpc.ApiCallable;
+import io.gapi.gax.grpc.PageDescriptor;
+import io.gapi.gax.grpc.ServiceApiSettings;
+import io.gapi.gax.internal.ApiUtils;
+import io.gapi.gax.protobuf.PathTemplate;
+import io.grpc.ManagedChannel;
+import java.io.IOException;
+
+// Manually-added imports: add custom (non-generated) imports after this point.
+
+
+
+// AUTO-GENERATED DOCUMENTATION AND SERVICE - see instructions at the top of the file for editing.
+/**
+ * See //google/logging/v2/logging.proto for documentation
+ *
+ * <!-- manual edit -->
+ * <!-- end manual edit -->
+ */
+@javax.annotation.Generated("by the veneer generator")
+public class ConfigServiceV2Api implements AutoCloseable {
+
+  // =========
+  // Constants
+  // =========
+
+  /**
+   * The default address of the service.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public static final String SERVICE_ADDRESS = "logging.googleapis.com";
+
+  /**
+   * The default port of the service.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public static final int DEFAULT_SERVICE_PORT = 443;
+
+
+  private static final ApiCallable<ListSinksRequest, ListSinksResponse>
+      LIST_SINKS = ApiCallable.create(ConfigServiceV2Grpc.METHOD_LIST_SINKS);
+  private static final ApiCallable<GetSinkRequest, LogSink>
+      GET_SINK = ApiCallable.create(ConfigServiceV2Grpc.METHOD_GET_SINK);
+  private static final ApiCallable<CreateSinkRequest, LogSink>
+      CREATE_SINK = ApiCallable.create(ConfigServiceV2Grpc.METHOD_CREATE_SINK);
+  private static final ApiCallable<UpdateSinkRequest, LogSink>
+      UPDATE_SINK = ApiCallable.create(ConfigServiceV2Grpc.METHOD_UPDATE_SINK);
+  private static final ApiCallable<DeleteSinkRequest, Empty>
+      DELETE_SINK = ApiCallable.create(ConfigServiceV2Grpc.METHOD_DELETE_SINK);
+
+  private static PageDescriptor<ListSinksRequest, ListSinksResponse, LogSink> LIST_SINKS_PAGE_DESC =
+      new PageDescriptor<ListSinksRequest, ListSinksResponse, LogSink>() {
+        @Override
+        public Object emptyToken() {
+          return "";
+        }
+        @Override
+        public ListSinksRequest injectToken(
+            ListSinksRequest payload, Object token) {
+          return ListSinksRequest
+            .newBuilder(payload)
+            .setPageToken((String) token)
+            .build();
+        }
+        @Override
+        public Object extractNextToken(ListSinksResponse payload) {
+          return payload.getNextPageToken();
+        }
+        @Override
+        public Iterable<LogSink> extractResources(ListSinksResponse payload) {
+          return payload.getSinksList();
+        }
+      };
+
+  private static String ALL_SCOPES[] = {
+    "https://www.googleapis.com/auth/logging.read",
+    "https://www.googleapis.com/auth/logging.write",
+    "https://www.googleapis.com/auth/logging.admin"
```

Comment:
This may not be practical to fix, because this is automated, and the knowledge of scope containment lives elsewhere. This should not do any harm and is not part of the public surface.